python/src/aiconfig/editor/client/src/components/AIConfigEditor.tsx

```diff
@@ -242,19 +242,21 @@ function AIConfigEditorBase({
       async (
         promptName: string,
         newPrompt: Prompt,
-        onSuccess: (aiconfigRes: AIConfig) => void,
-        onError: (err: unknown) => void
+        callbacks: {
+          onSuccess?: (aiconfigRes: AIConfig) => void;
+          onError?: (err: unknown) => void;
+        }
```

Comment:
what's the rationale to make this an object vs keep the callbacks defined?

Reply:
Using `onSuccess` and `onError` is a very common frontend pattern. I think Ryan is just following conventions?

Reply:
Ya, for js usually it's common to use an object when props can be undefined rather than having literal 'undefined' prop before a defined one